pkg/controller/tainteviction/taint_eviction.go

```diff
@@ -176,49 +182,124 @@ func getMinTolerationTime(tolerations []v1.Toleration) time.Duration {
+		getPodsAssignedToNode: func(nodeName string) ([]*v1.Pod, error) {
+			objs, err := podIndexer.ByIndex("spec.nodeName", nodeName)
+			if err != nil {
+				return nil, err
+			}
+			pods := make([]*v1.Pod, 0, len(objs))
+			for _, obj := range objs {
+				pod, ok := obj.(*v1.Pod)
+				if !ok {
+					continue
+				}
+				pods = append(pods, pod)
+			}
+			return pods, nil
+		},
+		taintedNodes: make(map[string][]v1.Taint),
+
+		nodeUpdateQueue: workqueue.NewWithConfig(workqueue.QueueConfig{Name: "noexec_taint_node"}),
+		podUpdateQueue:  workqueue.NewWithConfig(workqueue.QueueConfig{Name: "noexec_taint_pod"}),
+	}
+	tm.taintEvictionQueue = CreateWorkerQueue(deletePodHandler(c, tm.emitPodDeletionEvent, tm.name))
+
+	_, err := podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*v1.Pod)
+			tm.PodUpdated(nil, pod)
+		},
+		UpdateFunc: func(prev, obj interface{}) {
+			prevPod := prev.(*v1.Pod)
+			newPod := obj.(*v1.Pod)
+			tm.PodUpdated(prevPod, newPod)
+		},
+		DeleteFunc: func(obj interface{}) {
+			pod, isPod := obj.(*v1.Pod)
+			// We can get DeletedFinalStateUnknown instead of *v1.Pod here and we need to handle that correctly.
+			if !isPod {
+				deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+				if !ok {
+					logger.Error(nil, "Received unexpected object", "object", obj)
+					return
+				}
+				pod, ok = deletedState.Obj.(*v1.Pod)
+				if !ok {
+					logger.Error(nil, "DeletedFinalStateUnknown contained non-Pod object", "object", deletedState.Obj)
+					return
+				}
+			}
+			tm.PodUpdated(pod, nil)
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to add pod event handler: %w", err)
+	}
+
+	_, err = nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: controllerutil.CreateAddNodeHandler(func(node *v1.Node) error {
+			tm.NodeUpdated(nil, node)
+			return nil
+		}),
+		UpdateFunc: controllerutil.CreateUpdateNodeHandler(func(oldNode, newNode *v1.Node) error {
+			tm.NodeUpdated(oldNode, newNode)
+			return nil
+		}),
+		DeleteFunc: controllerutil.CreateDeleteNodeHandler(logger, func(node *v1.Node) error {
+			tm.NodeUpdated(node, nil)
+			return nil
+		}),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to add node event handler: %w", err)
 	}
-	tm.taintEvictionQueue = CreateWorkerQueue(deletePodHandler(c, tm.emitPodDeletionEvent))
 
-	return tm
+	return tm, nil
 }
 
-// Run starts NoExecuteTaintManager which will run in loop until `stopCh` is closed.
-func (tc *NoExecuteTaintManager) Run(ctx context.Context) {
+// Run starts the controller which will run in loop until `stopCh` is closed.
+func (tc *Controller) Run(ctx context.Context) {
 	defer utilruntime.HandleCrash()
 	logger := klog.FromContext(ctx)
-	logger.Info("Starting NoExecuteTaintManager")
+	logger.Info("Starting", "controller", tc.name)
+	defer logger.Info("Stopping", "controller", tc.name)
```

Comment:
```suggestion
	defer logger.Info(""Shutting down controller", "controller", tc.name)
```

Although I like `stopping` more, we should try to be consistent with the other controllers.